File: AtlasWeb/Core/Modules/Saglik_Lojistik/Stok_Modulleri/Sihhiye_Modulleri/Miad_Uzatim_Cikis_Modulu/ExtendExpDateOutFormViewModel.cs
//$452BEEED
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using TTDataDictionary;
using System.Collections.Generic;
using System.ComponentModel;

namespace Core.Controllers
{
    public partial class ExtendExpDateOutServiceController
    {
        partial void PostScript_ExtendExpDateOutForm(ExtendExpDateOutFormViewModel viewModel, ExtendExpDateOut extendExpDateOut, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            if (viewModel.TempOuttableLots != null)
            {
                foreach (TempOuttableLot tempOuttableLot in viewModel.TempOuttableLots)
                {

                    foreach (ExtendExpDateOutDetail extendExpirationDateDetailsImported in extendExpDateOut.ExtendExpDateOutDetails)
                    {

                        if (extendExpirationDateDetailsImported.OutExpirationDate != null)
                        {
                            BindingList<Stock> stocks = Stock.GetStoreMaterial(extendExpDateOut.ObjectContext, extendExpirationDateDetailsImported.StockAction.Store.ObjectID, extendExpirationDateDetailsImported.Material.ObjectID);
                            if (stocks.Count == 1)
                            {
                                Stock materialStock = stocks[0];
                                BindingList<StockTransaction.LOTOutableStockTransactions_Class> outableStockTransactions = StockTransaction.LOTOutableStockTransactions(materialStock.ObjectID);
                                foreach (StockTransaction.LOTOutableStockTransactions_Class outLot in outableStockTransactions)
                                {
                                    if (outLot.ExpirationDate == tempOuttableLot.ExpirationDate && outLot.LotNo == tempOuttableLot.LotNo)
                                    {
                                        OuttableLot outtableLot = new OuttableLot(extendExpDateOut.ObjectContext);
                                        outtableLot.LotNo = outLot.LotNo;
                                        if (outLot.ExpirationDate == null)
                                            outtableLot.ExpirationDate = DateTime.MinValue;
                                        else
                                            outtableLot.ExpirationDate = outLot.ExpirationDate;
                                        outtableLot.RestAmount = CurrencyType.ConvertFrom(outLot.Restamount);
                                        outtableLot.Amount = CurrencyType.ConvertFrom(outLot.Restamount);
                                        outtableLot.isUse = true;
                                        outtableLot.StockActionDetailOut = extendExpirationDateDetailsImported;
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }

        partial void AfterContextSaveScript_ExtendExpDateOutForm(ExtendExpDateOutFormViewModel viewModel, ExtendExpDateOut extendExpDateOut, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            string sonucMesaji = string.Empty;
            if (viewModel._ExtendExpDateOut.CurrentStateDefID == ExtendExpDateOut.States.Completed)
            {
                if (TTObjectClasses.SystemParameter.GetParameterValue("SENDTOMKYSFORAFTERCONTEXT", "TRUE") == "TRUE")
                {

                    sonucMesaji = extendExpDateOut.SendUpdateMessageToMKYS(Common.CurrentResource.MkysPassword);
                    TTUtils.InfoMessageService.Instance.ShowMessage(sonucMesaji);
                }
            }
        }
    }
}

namespace Core.Models
{
    public partial class ExtendExpDateOutFormViewModel: BaseViewModel
    {
        public List<TempOuttableLot> TempOuttableLots
        {
            get;
            set;
        }
    }
}

File: AtlasWeb/Core/Modules/Saglik_Lojistik/Stok_Modulleri/Sihhiye_Modulleri/Miad_Uzatim_Cikis_Modulu/ExtendExpDateOutFormViewModel.ts
//$A45A39CE
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { ExtendExpDateOut } from "NebulaClient/Model/AtlasClientModel";
import { ExtendExpDateOutDetail } from "NebulaClient/Model/AtlasClientModel";
import { StockActionSignDetail } from "NebulaClient/Model/AtlasClientModel";
import { Store } from "NebulaClient/Model/AtlasClientModel";
import { Supplier } from "NebulaClient/Model/AtlasClientModel";
import { Material } from "NebulaClient/Model/AtlasClientModel";
import { ResUser } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class ExtendExpDateOutFormViewModel extends BaseViewModel {
    @Type(() => ExtendExpDateOut)
    public _ExtendExpDateOut: ExtendExpDateOut = new ExtendExpDateOut();
    @Type(() => ExtendExpDateOutDetail)
    public ExtendExpDateOutDetailsGridList: Array<ExtendExpDateOutDetail> = new Array<ExtendExpDateOutDetail>();
    @Type(() => StockActionSignDetail)
    public StockActionSignDetailsGridList: Array<StockActionSignDetail> = new Array<StockActionSignDetail>();
    @Type(() => Store)
    public Stores: Array<Store> = new Array<Store>();
    @Type(() => Supplier)
    public Suppliers: Array<Supplier> = new Array<Supplier>();
    @Type(() => Material)
    public Materials: Array<Material> = new Array<Material>();
    @Type(() => ResUser)
    public ResUsers: Array<ResUser> = new Array<ResUser>();
    @Type(() => TempOuttableLot)
    public TempOuttableLots: Array<TempOuttableLot> = new Array<TempOuttableLot>();
}

export class TempOuttableLot {
    public Amount: Currency;
    public RestAmount: Currency;
    public ExpirationDate: Date;
    public LotNo: string;
    public isUse: boolean;
    public StockActionDetailOutID: string;
}